### DISCUSSION TITLE
Q&A regards owl-project

### DISCUSSION THREAD

**Discussion by bipul-mohanto** [owl-project/owl#155] [Q&A] [answered]
Hello!

Few general queries:

1. Is there any discussion forum (e.g., [NVidia Developer Discussion Forum](https://forums.developer.nvidia.com/), Discord, Reddit etc.) other than this `github` where the beginners like me can discuss about different problems? From google I could not find anything actually.
1. Besides the [OWL-Project github](https://owl-project.github.io/), [Ingo's Blog](https://ingowald.blog/2020/11/08/introducing-owl-a-node-graph-abstraction-layer-on-top-of-optix-7/), and the built-in `sample codes` come with OWL, is there any beginner friendly tutorial available? 
1. Is **OWL** support by `NVidia NSight`? Afaik OptiX is supported by `NVidia NSight` for GPU profiling (if I am not wrong).

Thanks a lot.

**Comment by ingowald** [comment, marked as answer]
Hey, Bipul. 

1) No, there is no official discussion forum, yet - as far as I understand github this here is exactly github's intended mechanism for that, it's just that a lot of the discussions between the main developers so far happened on slack or email, and external users only filed issues, never discussions yet. This here _should_ however be the perfect place to have those.

2) OWL intentionally contains a few samples directly in the codebase (owl/samples), and we intentionally created those to each be rather small and focussed on a given particular feature. So in a sense, they've been designed to server as "kind of" a tutorial, in particular first few parts that one by one show how to create context, frame buffer, geometry, etc. In addition there's also the optix7course that we did for siggraph a while back - that is't owl-specific, and more of a tutorial on _optix_ concepts, but the final code sample from that is also available in owl, ported to owl instead of using low-level optix codes. It would of course be nice to have even more of a "pure" tutorial, and I've actually thought about that a lot in the past - but that's surprisingly tricky to create right, because after the first few steps different users all wants another "next" step: say after chapter N on creating an accel struct over a triangle mesh one user wants to learn how to create instances, the other how to create usre geometry, while the next wants to learn more about tracing rays and shading, etc. If you do come up with some nice examples that you want to share, let us know!

3) it should; i personally only work on linux, but some of the other developers/users use it. if you run into issues, please file or re-open a github issue. I know that at some point owl didn't include the right lineinfo information for nsight to be really useful, but that has been fixed long ago.

**Reply by bipul-mohanto** [reply]
Hello Ingo Wald,

Personally I am a big fan of your work, and really motivated receiving a reply directly from you. Your **optix7course** tutorial series is really the finest tutorial on OptiX and CUDA. 

I have just a beginner in real-time ray tracing research. I am trying to achieve some result on varied sampling rate in path tracing to render the scene faster. Later I would try to extend the code base for Head Mounted Display if it works fine. Hope I could be benefited from the Project-OWL.
